libbeat/docs/queueconfig.asciidoc

```diff
@@ -76,3 +77,147 @@ will be immediately available for consumption.
 
 The default values is 0s.
 
+[float]
+[[configuration-internal-queue-spool]]
+=== Configure the file spool queue
+
+The file spooling queue stores all events in an on disk ring buffer. The spool
+has a write buffer, which new events are written to. Events written to the
+spool are forwarded to the outputs, only after the write buffer has been
+flushed successfully.
+
+The spool waits for the output to acknowledge or drop events. If the spool is
+full, no new events can be inserted. The spool will block. Space is freed only
+after a signal from the output has been received.
+
+On disk, the spool operates in pages. The `file.page_size` setting configures
+the files page size on creation time. The optimal page size depends on the
+effective block size, used by the underlying file system.
+
+This sample configuration enables the spool with all default settings and the
+default file path:
+
+[source,yaml]
+------------------------------------------------------------------------------
+queue.spool: ~
+------------------------------------------------------------------------------
+
```

Comment:
Might be worth pointing the user to the Configuration options section for the defaults.

Reply:
Added a link